xskillscore/core/probabilistic.py

```diff
@@ -526,73 +608,172 @@ def rps(
+    if member_dim not in forecasts.dims:
+        raise ValueError(
+            f"Expect to find {member_dim} in forecasts dimensions, found"
+            f"{forecasts.dims}."
         )
-
-    forecasts = histogram(
-        forecasts,
-        bins=[category_edges],
-        bin_names=bin_names,
-        dim=[member_dim],
-    )
     if fair:
-        e = forecasts
+        M = forecasts[member_dim].size
+
+    forecasts = _bool_to_int(forecasts)
+
+    _check_identical_xr_types(observations, forecasts)
+
+    # different entry point of calculating RPS based on category_edges
+    # category_edges tuple of two: use for obs and forecast category_edges separately
+    if isinstance(category_edges, (tuple, np.ndarray, xr.DataArray, xr.Dataset)):
+        if isinstance(category_edges, tuple):
+            assert isinstance(category_edges[0], type(category_edges[1]))
+            observations_edges, forecasts_edges = category_edges
+        else:  # category_edges only given once, so use for both obs and forecasts
+            observations_edges, forecasts_edges = category_edges, category_edges
+
+        if isinstance(observations_edges, np.ndarray):
+            # convert category_edges as xr object
+            observations_edges = xr.DataArray(
+                observations_edges,
+                dims="category_edge",
+                coords={"category_edge": observations_edges},
+            )
+            observations_edges = xr.ones_like(observations) * observations_edges
 
-    # normalize f.sum()=1
-    forecasts = forecasts / forecasts.sum(bin_dim)
-    observations = observations / observations.sum(bin_dim)
+            forecasts_edges = xr.DataArray(
+                forecasts_edges,
+                dims="category_edge",
+                coords={"category_edge": forecasts_edges},
+            )
+            forecasts_edges = (
+                xr.ones_like(
+                    forecasts
+                    if member_dim not in forecasts.dims
+                    else forecasts.isel({member_dim: 0}, drop=True)
+                )
+                * forecasts_edges
+            )
 
-    Fc = forecasts.cumsum(bin_dim)
-    Oc = observations.cumsum(bin_dim)
+        _check_identical_xr_types(forecasts_edges, forecasts)
+        _check_identical_xr_types(observations_edges, forecasts)
 
-    if fair:
-        Ec = e.cumsum(bin_dim)
-        res = (((Ec / M) - Oc) ** 2 - Ec * (M - Ec) / (M ** 2 * (M - 1))).sum(bin_dim)
+        _check_data_within_edges(forecasts, forecasts_edges)
+        _check_data_within_edges(observations, observations_edges)
+
+        # cumulative probs, ignore lowest threshold as below category_edges
+        # ignores the right-most edge. The effective right-most edge is np.inf.
+        # therefore the CDFs Fc and Oc both reach 1 for the right-most edge.
+        Fc = (
+            (forecasts < forecasts_edges)
+            .mean(member_dim)
+            .isel({bin_dim: slice(1, None)})
```

Comment:
Perhaps it's better to be clear about how this operation is done in the docstring (see above), rather than to silently ignore the lowest threshold?

Reply:
I wouldnt say we silently ignore the lower threshold. we compute `<` and therefore throw the lower away because we dont need it, i.e. outside of category_edge